### PR TITLE
feat: add LinkedIn OIDCProvider

### PR DESCRIPTION
**Pull request by J0**
## What kind of change does this PR introduce?

We introduce a LinkedIn (OIDC) provider and plan to deprecate the existing LinkedIn provider. To that end, we have greyed out the LinkedIn provider(except for toggle) in hopes of encouraging developers to move over from LinkedIn (Deprecated) to LinkedIn (OIDC)

Supports: https://github.com/supabase/infrastructure/pull/15177

- [x] Need to update the corresponding docs

### CODE REVIEW

**Review by kangmingtay — APPROVED** [review]
Can we also add that developers would need to include the following scopes on the linkedin dev platform:
image

Other than that, it LGTM from the docs side of things

**Comment by J0** [comment]
Oh I think it's actually included when one [requests access for Sign In with LinkedIn OIDC](https://github.com/supabase/supabase/pull/18225/files#diff-f6321d8c93f1663af3962854ede0fd9f6c59fe79a40b451d3f31b965601cb73cR46)  - I think it's a good idea to remind devs to check though. Have added it in.

File: studio/components/interfaces/Auth/AuthProvidersForm/ProviderForm.tsx [hunk 1]
```diff
@@ -139,39 +146,45 @@ const ProviderForm = ({ config, provider }: ProviderFormProps) => {
...
-                        Linkedin has updated their OAuth APIs, which has broken the current Supabase
-                        implementation. We're actively tracking{' '}
-                        <a
-                          className="underline"
-                          href="https://github.com/supabase/gotrue/issues/1216"
-                        >
-                          this issue
-                        </a>{' '}
-                        and you can subscribe to it for future updates.
+                        As of 1st August, LinkedIn has updated their OAuth API scopes. Please usethe
```
Comment (J0) [inline studio/components/interfaces/Auth/AuthProvidersForm/ProviderForm.tsx:159]:
```suggestion
                        As of 1st August, LinkedIn has updated their OAuth API scopes. Please use the
```

File: studio/components/interfaces/Auth/AuthProvidersForm/ProviderForm.tsx [hunk 2]
```diff
@@ -139,39 +146,45 @@ const ProviderForm = ({ config, provider }: ProviderFormProps) => {
...
+                        new LinkedIn provider below. Developers using this provider should move over
+                        to the new provider. Please refer to our docs (insert link) for more
```
Comment (J0) [inline studio/components/interfaces/Auth/AuthProvidersForm/ProviderForm.tsx:161]:
```suggestion
                        to the new provider. Please refer to our [docs](/docs/pages/guides/auth/social-login/auth-linkedin.mdx) for more
```

**Review by ivasilov — APPROVED** [review]

**Review by alaister — APPROVED** [review]

**Pull request merged**